### ISSUE TITLE
Include license

### ISSUE THREAD

**Issue by Juanlu001** [OpenFortranProject/ofp-sdf#8]
I am unable to find the license of the project. Former ANTLR-based project used BSD and EPL according to SourceForge, but it would be nice to clarify for this new incarnation of the compiler.

**Comment by rouson** [comment]
@rasmussn, would it be ok to add the [license](http://www.sourceryinstitute.org/license.html) that OpenCoarrays uses, replacing the named copyright holders with whatever names you provide?  The OpenCoarrays license is a 3-clause BSD license chosen because it increases the likelihood that corporations will use the software.

**Comment by rasmussn** [comment]
Yes, I was planning on using yours with the copyright modified.  I've
created the attached file.  Perhaps you could look it over and let me know
where to put it.  I will also begin to put it on individual files where
possible.

Thanks,
Craig


On Mon, Feb 15, 2016 at 8:00 PM, Damian Rouson <notifications@github.com>
wrote:

> @rasmussn <https://github.com/rasmussn>, would it be ok to add the license
> <http://www.sourceryinstitute.org/license.html> that OpenCoarrays uses,
> replacing the named copyright holders with whatever names you provide? The
> OpenCoarrays license is a 3-clause BSD license chosen because it increases
> the likelihood that corporations will use the software.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenFortranProject/ofp-sdf/issues/8#issuecomment-184507506>
> .
>

/*
 * Open Fortran Project (OFP)
 *
 * All files in this project are released under the following license:
 *
 * OFP is distributed under the OSI-approved BSD 3-clause License:
 * Copyright (c) 2013-2016, Paradox PTF
 * Copyright (c) 2013-2016, University of Oregon
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without modification,
 * are permitted provided that the following conditions are met:
 *
 * 1. Redistributions of source code must retain the above copyright notice, this
 *    list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright notice, this
 *    list of conditions and the following disclaimer in the documentation and/or
 *    other materials provided with the distribution.
 * 3. Neither the names of the copyright holders nor the names of their contributors
 *    may be used to endorse or promote products derived from this software without
 *    specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
 * IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
 * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
 * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
 * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
 * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGE.
 */

**Comment by rouson** [comment]
Looks fine. I would put it in a file named “LICENSE” in the top level and, as you state, at the top of individual files too.

I believe @zbeekman is cooking up some sort of magic to help keep the copyright dates current.  I think it will be in the form of a git hook that will check the copyright date on files as part of the automatic tests that get run every a pull request is submitted.

Damian

**Comment by rasmussn** [comment]
Why don't I add the licenses to the individual files.  Some of the files
don't allow comments :-(

-craig

**Comment by zbeekman** [comment]
@rasmussn I think all you really need to do is add a file named `LICENSE` in the git projects root directory containing the appropriate license text and commit that. Then in files allowing comments you can include a shorter copyright notice, and refer the reader to the top level `LICENSE` file.

@rouson As far as the git-hook goes, I'll look into it. No promises, though.

**Comment by rouson** [comment]
@zbeekman.  I like that suggestion.  We can adopt the same for OpenCoarrays at some point.  That would decrease the hassle of maintaining dates.  The date would only need to be updated annually the top-level license file. Also, I just noticed that we have separate top-level COPYRIGHT and LICENSE files, which essentially match in content but not in format.  I'll delete the COPYRIGHT file.

@zbeekman just in case it's ever useful, a generic version of the license file is [online](http://www.sourceryinstitute.org/license.html).

**Comment by zbeekman** [comment]
> @zbeekman. I like that suggestion. We can adopt the same for OpenCoarrays at some point. That would decrease the hassle of maintaining dates. The date would only need to be updated annually the top-level license file. Also, I just noticed that we have separate top-level COPYRIGHT and LICENSE files, which essentially match in content but not in format. I'll delete the COPYRIGHT file.

I think wherever it's possible to include the "not our fault, use at your own risk" indemnification/disclaimer clause (the final ALL CAPS paragraph of the license/disclaimer) and some indication of Copyright it's probably a good idea, even if you have to update the copyright years etc. This makes it harder for people to rip off your work without attribution and may help prevent legal problems<sup>I'm no<sup>lawyer<sup>though</sup></sup></sup> should someone grab a component piece of software and attempt to use it for something.

> @zbeekman just in case it's ever useful, a generic version of the license file is online.

Yes, it's also available from https://opensource.org/licenses/BSD-3-Clause and if you create a new repository on github.com you can choose to have it initialized with any number of open licenses, .gitignore files, and README.md boiler plates.
